include/roboteam_ai/utilities/Dealer.h

```diff
@@ -57,13 +57,14 @@ class Dealer {
     struct RoleInfo {
         DealerFlagPriority priority;
         std::vector<DealerFlag> flags;
+        int forcedID = -1;
     };
 
     using FlagMap = std::map<std::string, RoleInfo>;
     Dealer(v::WorldDataView world, rtt::world::Field *field);
     virtual ~Dealer() = default;  // needed for test
     // Create a distribution of robots
-    std::unordered_map<std::string, v::RobotView> distribute(const std::vector<v::RobotView> &allRobots, const FlagMap &flagMap,
+    std::unordered_map<std::string, v::RobotView> distribute(const std::vector<v::RobotView> &_allRobots, FlagMap &flagMap,
```

Comment:
why are you modifying the FlagMap here?

Reply:
I was not sure how to do it nicely, however the thinking was that:
if you force a role on a Robot, the role and robot should not be considered anymore. By removing them, it would mean that the scoreMatric and everything after that does not consider the Role and Robot, which is more efficient.

The downside of such is that the flagMap will be shorter than in the Play. However the flag map is not used after the dealer anymore. Alternatively, i could make a copy of flagMap and change that, but as it was not used anymore, i chose the not make copies where it was not needed

Reply:
The thing is that the function is called "distribute", which is now modifying things, and has side effects. 

It's fine but change the name of the function to something that makes it clear it modifies.

Reply:
I'd prefer then to just make a copy, removing the reference to the flagMap (so that in play has the original flagMap and the dealer can do with it as it wants)